src/main.cpp

```diff
@@ -3866,9 +3866,14 @@ bool CBlock::CheckBlock(std::string sCaller, int height1, int64_t Mint, bool fCh
     double dAccrualAge = 0;
     double dMagnitudeUnit = 0;
     double dAvgMagnitude = 0;
-    int64_t nCoinAge = 0;
+    uint64_t nCoinAge = 0;
     int64_t nFees = 0;
 
+    // Check CoinAge - Bug fix
+    CTxDB txdb;
+    if (!vtx[1].GetCoinAge(txdb, nCoinAge))
+        return error("CheckBlock[] : %s unable to get coin age for coinstake", vtx[1].GetHash().ToString().substr(0,10).c_str());
+
```

Comment:
Do not calculate coinage in CheckBlock. transactions are not stable at this point. And interest is not checked in this function anyway, only OUT_POR, if you scroll down a bit. Interest is checked in connectblock. You can edit the debug outputs to clarify.

Reply:
ah i see we only check POR then why is the coinage , etc even put in there. makes no sense to even have the debug outputs with it either GetProofOfStakeReward does not provide coinage info and im not seeing connect block below that what line numebr since function ends on 4032

Reply:
By below I mean: `if (bb.ResearchSubsidy > ((OUT_POR*1.25)+1))`, it checks only PoR.
CheckBlock() is [higher up here](https://github.com/gridcoin/Gridcoin-Research/pull/677/files#diff-7ec3c68a81efff79b6ca22ac1f1eabbaL2841).

Reply:
yea i see that now but why we even put the OUT_INTEREST in the debug output there. seems pointless regardless and can be misleading. i'll deal with this tmr i got to take son to preschool in morning

Reply:
I agree, change the log to only show what matters.